src/OpenID/Connect/Client/Authentication.hs

```diff
@@ -49,7 +49,7 @@ import OpenID.Connect.JSON
 applyRequestAuthentication
   :: forall m. MonadRandom m
   => Credentials                -- ^ Client credentials.
-  -> [ClientAuthentication]     -- ^ Available authentication methods.
+  -> [TokenEndpointAuthMethod]  -- ^ Available authentication methods.
```

Comment:
I was hoping that this function wouldn't change its type signature as a signal to users that only standard (non-extension) authentication methods are supported.  Does that cause a problem somewhere else?